src/charm.py

```diff
@@ -1,104 +1,432 @@
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
-                    "startup": "enabled",
-                    "environment": {"thing": self.model.config["thing"]},
+                    "summary": "sdlc",
+                    "command": (
+                        # NOTE(aznashwan): starting through bash is required
+                        # for the classpath glob (-cp ...) to be expanded:
+                        "/bin/sh -c 'java -XX:+ExitOnOutOfMemoryError "
+                        "-XX:MaxRAMPercentage=60 -Xss4M -cp /app/bin/*.jar "
+                        "-Dfile.encoding=UTF8 "
+                        "org.finos.legend.sdlc.server.LegendSDLCServer "
+                        "server %s'" % (
+                            SDLC_CONFIG_FILE_CONTAINER_LOCAL_PATH)
+                    ),
+                    # NOTE(aznashwan): considering the SDLC service expects
+                    # a singular config file which already contains all
+                    # relevant options in it (some of which will require the
+                    # relation with DB/Gitlab to have already been
+                    # established), we do not auto-start:
+                    "startup": "disabled",
+                    # TODO(aznashwan): determine any env vars we could pass
+                    # (most notably, things like the RAM percentage etc...)
+                    "environment": {},
                 }
             },
         }
+
         # Add intial Pebble config layer using the Pebble API
-        container.add_layer("httpbin", pebble_layer, combine=True)
-        # Autostart any services that were defined with startup: enabled
-        container.autostart()
-        # Learn more about statuses in the SDK docs:
-        # https://juju.is/docs/sdk/constructs#heading--statuses
-        self.unit.status = ActiveStatus()
-
-    def _on_config_changed(self, _):
-        """Just an example to show how to deal with changed configuration.
-
-        TEMPLATE-TODO: change this example to suit your needs.
-        If you don't need to handle config, you can remove this method,
-        the hook created in __init__.py for it, the corresponding test,
-        and the config.py file.
-
-        Learn more about config at https://juju.is/docs/sdk/config
+        container.add_layer("sdlc", pebble_layer, combine=True)
+
+        # NOTE(aznashwan): as mentioned above, we will *not* be auto-starting
+        # the service until the relations with DBMan and Gitlab are added:
+        # container.autostart()
+
+        self.unit.status = model.BlockedStatus(
+            "Awaiting Legend Database and Gitlab relations.")
+
+    def _get_logging_level_from_config(self, option_name) -> str:
+        """Fetches the config option with the given name and checks to
+        ensure that it is a valid `java.utils.logging` log level.
+
+        Returns None if an option is invalid.
```

Comment:
We generally use the Google docstring format, which would look like so:

```suggestion
        ensure that it is a valid `java.utils.logging` log level.
        
        Args:
            option_name: string, name of the config option to fetch from charm config

        Returns:
            a string value containing the log level. An empty string is returned if config is invalid
```